main.go

```diff
@@ -615,6 +615,12 @@ func (context *Context) serveStatus() error {
 		mux.Handle("/debug/pprof/trace", http.HandlerFunc(pprof.Trace))
 	}
 
+	serveHTTPS := true
+	if strings.HasPrefix(*statusAddress, "http://") {
+		*statusAddress = (*statusAddress)[7:]
```

Comment:
(1) Would prefer that we avoid changing the global flag value here, as it may lead to odd side-effects if the value is ever used somewhere else. Let's introduce a local var for the status address. 
(2) This code should also handle the case where the prefix might be `https://` instead of `http://`, and behave accordingly. Ideally create a function that parses the status address and returns `(https bool, network, address string, err error)` based on parsing net/url and socket.ParseAddress.

Reply:
@csstaub Thanks for the comment.

Pushed a new commit, but instead of a function return (https bool, network, address string, err error), I created socket.ParseHTTPAddress function that return (https bool, address string)

the return address could be further parsed by existing socket.ParseAddress.

IMHO, this would avoid more code changes.